system/reports/assets/simple.cfm

```diff
@@ -309,32 +311,52 @@ code {
 	<cfreturn bootstrapClass>
 </cffunction>
 
+<cffunction name="statusToEmoji" output="false">
+	<cfargument name="status">
+	<cfargument name="emojiService">
+
+	<cfset emoji="">
+
+	<cfif lcase( arguments.status ) eq "failed">
+		<cfset emoji=arguments.emojiService.get( 'warning' )>
+	<cfelseif lcase( arguments.status ) eq "error">
+		<cfset emoji=arguments.emojiService.get( 'x' )>
+	<cfelseif lcase( arguments.status ) eq "passed">
+		<cfset emoji=arguments.emojiService.get( 'white_check_mark' )>
+	<cfelseif lcase( arguments.status ) eq "skipped">
+		<cfset emoji=arguments.emojiService.get( 'no_entry' )>
+	</cfif>
+
+	<cfreturn emoji>
+</cffunction>
+
 <!--- Recursive Output --->
 <cffunction name="genSuiteReport" output="false">
 	<cfargument name="suiteStats">
 	<cfargument name="bundleStats">
+	<cfargument name="emojiService">
```

Comment:
Why pass this, when the emojiService is already in the `local` scope.  Just use it.

Reply:
I though the same, but it was not accessible inside the functions. If I removed `var` from its declaration it was accessible, but then it would be in the `variables` scope. I was surprised I could not access it. It might have to do with the fact that they are functions nested inside another function. Although they would be in the scope of the parent function and should have access to variable declared in it. It might be a Lucee bug or maybe I am trying to follow the scoping rules of JavaScript and they don't apply here.